### DISCUSSION TITLE
Please, add dotnet appsettings.json scanning support

### DISCUSSION THREAD

**Discussion by mykolaichuk** [aquasecurity/trivy#5407] [Q&A] [answered]
### Description

Please, add dotnet [appsettings.json](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0#appsettingsjson) scanning support as far as this config files can also include secrets like DB connectionstrings and so on.

### Target

Filesystem

### Scanner

Secret

**Comment by knqyf263** [comment]
Trivy scans all plain-text files for secrets.

**Reply by mykolaichuk** [reply]
Hm, then why it didn't find anything in my appsettings.json where I'm sure secrets were present? Should I write a custom rule for that then?

I've put an Azure Storage Account connection string for testing purposes.

**Reply by knqyf263** [reply]
All rules are listed here. If you find something missing, we welcome your contribution.
https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/secret/builtin-rules.go

**Reply by mykolaichuk** [reply, marked as answer]
I see now, thanks. There are no rules for Azure connection strings, secrets, etc.

And what is the best way to search for secrets in appsettings.json? They are stored in a key-value pairs, like
```json
{
    "Key": "mykey",
    "Password": "mypassword",
}
```

**Reply by DmitriyLewen** [reply]
Hello @mykolaichuk
You can create your own rule - https://aquasecurity.github.io/trivy/v0.46/docs/scanner/secret/#custom-rules:
- only for `appsettings.json` files.
- with a regexp that will help you find your secrets.

**Reply by mykolaichuk** [reply]
Ok, got you! Thanks a lot @knqyf263 @DmitriyLewen !
